src/queryBuilder/QueryBuilder.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react'
+import * as F from 'futil-js'
+import _ from 'lodash/fp'
+import { observable } from 'mobx'
+import { Provider } from 'mobx-react'
+import DDContext from './DragDrop/DDContext'
+import { Component } from '../utils/mobx-react-utils'
+import Group from './Group'
+import styles from '../styles'
+import { oppositeJoin } from '../utils/search'
+import treeUtils from '../utils/tree'
+let { background } = styles
+let randomString = () =>
+  Math.random()
+    .toString(36)
+    .substring(7)
+
+let { encode } = F.encoder('/') // todo get from client
+
+let blankNode = () => ({ key: randomString() })
+let replaceOn = (list, from, to) => list.splice(list.indexOf(from), 1, to)
+
+let ContextureClientBridge = (Types, Tree) => {
+  // not sure why reassignment is needed - its 2am and that makes it work... fix tomorrow!
```

Comment:
Is this comment still relevant?

Reply:
Yes 😞

Reply:
It is a mystery because it works in the unit test:

https://github.com/smartprocure/contexture-client/blob/master/test/index.js#L631-L641

and as you can see it is doing an extendOn which mutates: https://github.com/smartprocure/contexture-client/blob/master/src/index.js#L112-L116

Reply:
This is fixed by: https://github.com/smartprocure/contexture-client/pull/48